### PR TITLE
chore: refactor inference service

### PR DESCRIPTION
**Pull request by vansangpfiev**
## Describe Your Changes

- call inference service API instead of http request for start stop model

## Fixes Issues

- Closes #
- Closes #

## Self Checklist

- [ ] Added relevant comments, esp in complex areas
- [ ] Updated docs (for bug fixes / features)
- [ ] Created issues for follow-up changes or refactoring needed

### CODE REVIEW

File: engine/services/inference_service.h [hunk 1]
```diff
@@ -2,7 +2,9 @@
 
 #include <optional>
 #include <variant>
-#include "common/base.h"
+#include <mutex>
+#include <queue>
+#include <condition_variable>
```
Comment (namchuai) [inline engine/services/inference_service.h]: Do we use this import?

**Review by namchuai — APPROVED** [review]
✅ lgtm

File: engine/services/inference_service.h (hunk 1 above)
Comment (vansangpfiev) [reply engine/services/inference_service.h]: No - we don't

File: engine/controllers/server.cc [hunk 2]
```diff
@@ -95,15 +96,15 @@ void server::GetModels(const HttpRequestPtr& req,
...
+  auto ir = inference_svc_->GetEngines(req->getJsonObject());
   auto resp = cortex_utils::CreateCortexHttpJsonResponse(ir);
   callback(resp);
 }
 
 void server::FineTuning(
     const HttpRequestPtr& req,
     std::function<void(const HttpResponsePtr&)>&& callback) {
-  auto ir = inference_svc_.FineTuning(req->getJsonObject());
+  auto ir = inference_svc_->FineTuning(req->getJsonObject());
```
Comment (dan-menlo) [inline engine/controllers/server.cc:107]: @vansangpfiev Can I check: why do we have a FineTuning endpoint, if we haven't implemented it yet?

Comment (vansangpfiev) [reply engine/controllers/server.cc:107]: This is the legacy code from last time when we were trying to integrate `cortex.python` into `cortex.cpp`

**Pull request merged**